ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java

```diff
@@ -464,8 +465,21 @@ private HelixBootstrapUpgradeUtil(String hardwareLayoutPath, String partitionLay
         throw new IllegalArgumentException(
             entry.getKey() + " has invalid number of ZK endpoints: " + zkConnectStrs.size());
       }
-      HelixAdmin admin = helixAdminFactory.getHelixAdmin(zkConnectStrs.get(0));
+      HelixAdmin admin;
+      RealmAwareZkClient zkClient = null;
+      if (helixAdminFactory == null) {
+        RealmAwareZkClient.RealmAwareZkClientConfig clientConfig =
+            new RealmAwareZkClient.RealmAwareZkClientConfig().setConnectInitTimeout(30 * 1000L)
+                .setZkSerializer(new ZNRecordSerializer());
```

Comment:
The ZNRecordSerializer, HelixZkClient and ZKHelixAdmin are deprecated. Maybe add a to-do to update in the future?

Reply:
sure, will do